vmdb/app/models/ems_refresh/save_inventory_helper.rb

```diff
@@ -32,23 +34,19 @@ def save_inventory_prep(find_key, child_keys, extra_keys)
     return find_key, child_keys, extra_keys, remove_keys
   end
 
-  def save_inventory(type, klass, parent, hash, deletes, new_records, record_index, record_index_columns, find_key, child_keys, remove_keys)
-    # Backup keys that cannot be written directly to the database
-    key_backup = backup_keys(hash, remove_keys)
-
+  def save_inventory(type, parent, hash, deletes = nil, new_records = nil, record_index = nil,
+                     record_index_columns = nil, find_key = nil)
     # Find the record, and update if found, else create it
     found = find_key.blank? ? parent.send(type) : self.save_inventory_record_index_fetch(record_index, record_index_columns, hash, find_key)
     if found.nil?
-      found = klass.new(hash)
-      new_records.nil? ? parent.send("#{type}=", found) : new_records << found
+      found = find_key ? parent.send(type).build(hash) : parent.send("build_#{type}", hash)
+      # TODO: these should not be necessary. Probably missing some parent.save calls
```

Comment:
@Fryguy cool - you said a few things that triggered some thoughts for me.

> The `(true)` was to ensure that the in memory cache was ignored in case it was cached differently ahead of time.

I wonder why we need to invalidate the caches. Feels like this is just there to mask another problem.

> The push is nice because it writes the batch to the database immediately if the parent is already saved.

We are on the same page here. If anything, I was trying encourage more batching. Want to make sure we call `parent.save` at the top most parent, and then leverage that more.

> but saving the parent doesn't save updated children.

Do we want to introduce [`autosave: true`](http://apidock.com/rails/v3.2.13/ActiveRecord/AutosaveAssociation) to fix this? /cc @tenderlove
This feels like we could use active record more, removing the extra `update_attributes` calls and move more of the transactions into the `parent.save` statement.

You know, we may want to also introduce [`mark_for_destruction`](http://apidock.com/rails/ActiveRecord/AutosaveAssociation/mark_for_destruction).